test/idp/src/main/kotlin/org/codice/compliance/web/sso/PostLoginTest.kt

```diff
@@ -71,12 +76,15 @@ class PostLoginTest : StringSpec() {
                     .`when`()
                     .post(Common.getSingleSignOnLocation(SamlProtocol.POST_BINDING))
 
-            response.statusCode shouldBe HTTP_OK
-            val idpResponse = getServiceProvider(IdpResponder::class.java).getIdpPostResponse(response).apply {
-                isRelayStateGiven = true
-            }
+            response.statusCode shouldBe 200
+            val idpResponse = getServiceProvider(IdpResponder::class.java)
+                    .getIdpPostResponse(response).decorate().apply {
+                        isRelayStateGiven = true
+                    }
+
+            idpResponse.bindingVerifier().verify()
 
-            val responseDom = verifyResponse(idpResponse)
+            val responseDom = idpResponse.responseDom!!
```

Comment:
❗️ This can potentially through an NPE (protection against which is, please recall, one of the primary advantages that Kotlin provides over Java).

See note on `IdpResponseDecorators.kt`.